sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -906,7 +906,7 @@ case class AssertNotNull(child: Expression, walkedTypePath: Seq[String])
   override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val childGen = child.genCode(ctx)
 
-    val errMsgField = ctx.addReferenceObj("errMsg", errMsg)
+    val errMsgField = ctx.addReferenceObj(errMsg)
```

Comment:
thanks - then i'd actually add a comment in the caller indicating it is important to call the variant that doesn't create a local field.